src/main/java/io/mosip/iiitb/odte/lib/ApiRequestService.java

```diff
@@ -0,0 +1,281 @@
+        return tokenId;
+    }
+
+    /**
+     * @param authToken = Authorization Token set in cookie header
+     * @param uid = VID | UIN
+     * @param issuer = auth partner id
+     * @param
+     * @return
+     */
+    public String getCredentialRequestId(
+            String authToken,
+            String uid,
+            String issuer,
+            CredentialRequestAdditionalDataDto additionalData
+    ) throws IOException, InterruptedException {
+        URI url = baseUri.resolve(credentialRequestEndpoint);
+        HttpCookie cookie = getAuthCookie(authToken);
+        Map<String, Object> authRequest = new HashMap<>();
+        authRequest.put("id", uid);
+        authRequest.put("recipient", authRecipient);
+        authRequest.put("issuer", issuer);
+        authRequest.put("credentialType", authCredentialType);
+        authRequest.put("user", CREDENTIAL_REQUEST_GENERATOR_USER);
+        authRequest.put("encrypt", false); //set property
+        authRequest.put("sharableAttributes", null);
+        authRequest.put("additionalData", additionalData);
+
+        Map<String, Object> body = new HashMap<>();
+        body.put("id", "mosip.credentialrequest");
+        body.put("request", authRequest);
+        body.put("requestTime", getTimeStamp());
+        body.put("version", "1.0");
+
+        HttpRequester.ResponseWrapper<CredentialRequestGeneratorRawResponseDto> httpResponse = httpRequester.makePostRequest(
+            url.toString(),
+            body,
+            cookie,
+            CredentialRequestGeneratorRawResponseDto.class
+        );
+        CredentialRequestGeneratorResponseDto response = httpResponse.getBody().getResponse();
+        String requestId = response.getRequestId();
+        return requestId;
+    }
+
+    public IssueCredentialsRawResponseDto issueCredentials(
+        String authToken,
+        String id,
+        String issuer,
+        String requestId,
+        CredentialRequestAdditionalDataDto additionalData
+    ) throws IOException, InterruptedException {
+        Map<String, Object> body = new HashMap<>();
+        body.put("id", id);
+        body.put("credentialType", authCredentialType);
+        body.put("issuer", issuer);
+        body.put("requestId", requestId);
+        body.put("additionalData", additionalData);
+        URI uri = baseUri.resolve(CredentialIssuanceEndpoint);
+        HttpRequester.ResponseWrapper<IssueCredentialsRawResponseDto> httpResponse = httpRequester.makePostRequest(
+            uri.toString(),
+            body,
+            getAuthCookie(authToken),
+            IssueCredentialsRawResponseDto.class
+        );
+        IssueCredentialsRawResponseDto responseBody = httpResponse.getBody();
+        return responseBody;
+    }
+
+
+    public String getAuthToken(String appId, String clientId, String clientPass)
+            throws IOException, InterruptedException {
+        if (authTokenCache.getValue() != null)
```

Comment:
there should also be a way to invalidate the cache when there is a 401 error

Reply:
Done, now it invalidates on 401 response